### PR TITLE
OCM-8986 | test: Automated id:68173,66874,67057

### PR DESCRIPTION
**Pull request by xueli181114**

### CODE REVIEW

**Comment by xueli181114** [comment]
```
lixue@Xue-Lis-MacBook-Pro rosa % ginkgo -focus "(68173|66874|67057)" tests/e2e
Running Suite: ROSA CLI e2e tests suite - /Users/lixue/Workspace/rosa/tests/e2e
===============================================================================
Random Seed: 1719474903

Will run 3 of 119 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSStime="2024-06-27 15:55:24" level=info msg="Trying to list subnets of the vpc"
time="2024-06-27 15:55:24" level=info msg="Got 6 subnets"
time="2024-06-27 15:55:24" level=info msg="subnet-050e8e0cf96f10d85\t10.0.4.0/24\tus-west-2c\t"
time="2024-06-27 15:55:24" level=info msg="subnet-0a69c2da274bf20f1\t10.0.3.0/24\tus-west-2b\t"
time="2024-06-27 15:55:24" level=info msg="subnet-016492bd22837a647\t10.0.5.0/24\tus-west-2c\t"
time="2024-06-27 15:55:24" level=info msg="subnet-0b4f00f3560a65c23\t10.0.1.0/24\tus-west-2a\t"
time="2024-06-27 15:55:24" level=info msg="subnet-0091e54c2d71097d4\t10.0.2.0/24\tus-west-2b\t"
time="2024-06-27 15:55:24" level=info msg="subnet-03cbf4f8fc135c687\t10.0.0.0/24\tus-west-2a\t"
time="2024-06-27 15:55:24" level=info msg="Got custom rt rtb-06a89945a2a9364bc "
time="2024-06-27 15:55:24" level=info msg="Got custom rt rtb-0fb33566215203dc4 "
time="2024-06-27 15:55:24" level=info msg="Got custom rt rtb-02d9997529bf6c8d4 "
time="2024-06-27 15:55:24" level=info msg="Got custom rt rtb-05f4e9044d2479887 "
time="2024-06-27 15:55:24" level=info msg="Got custom rt rtb-08228ba929bc76b36 "
time="2024-06-27 15:55:24" level=info msg="Got custom rt rtb-007c5038a42f24474 "
time="2024-06-27 15:55:24" level=info msg="Got main association for rt rtb-0b7b1de0c7c905b0b"
time="2024-06-27 15:55:25" level=info msg="Create security group sg-044700bb18e0be9a4 success for vpc-0e2644ed2c3bc215e"
time="2024-06-27 15:55:26" level=info msg="Tag resource sg-044700bb18e0be9a4 successfully"
time="2024-06-27 15:55:26" level=info msg="Created tagged security group with ID sg-044700bb18e0be9a4"
time="2024-06-27 15:55:26" level=info msg="Authorize security group success sg-044700bb18e0be9a4"
time="2024-06-27 15:55:26" level=info msg="Authorize security group success sg-044700bb18e0be9a4"
time="2024-06-27 15:55:27" level=info msg="Create security group sg-0b344cc95f231b66f success for vpc-0e2644ed2c3bc215e"
time="2024-06-27 15:55:27" level=info msg="Tag resource sg-0b344cc95f231b66f successfully"
time="2024-06-27 15:55:27" level=info msg="Created tagged security group with ID sg-0b344cc95f231b66f"
time="2024-06-27 15:55:28" level=info msg="Authorize security group success sg-0b344cc95f231b66f"
time="2024-06-27 15:55:28" level=info msg="Authorize security group success sg-0b344cc95f231b66f"
time="2024-06-27 15:55:29" level=info msg="Create security group sg-09ed3a9ea9c1ab834 success for vpc-0e2644ed2c3bc215e"
time="2024-06-27 15:55:29" level=info msg="Tag resource sg-09ed3a9ea9c1ab834 successfully"
time="2024-06-27 15:55:29" level=info msg="Created tagged security group with ID sg-09ed3a9ea9c1ab834"
time="2024-06-27 15:55:30" level=info msg="Authorize security group success sg-09ed3a9ea9c1ab834"
time="2024-06-27 15:55:31" level=info msg="Authorize security group success sg-09ed3a9ea9c1ab834"
time="2024-06-27 15:55:57" level=info msg="Release rules successfully for SG sg-044700bb18e0be9a4"
time="2024-06-27 15:55:57" level=info msg="Delete security group sg-044700bb18e0be9a4 success "
time="2024-06-27 15:55:58" level=info msg="Release rules successfully for SG sg-0b344cc95f231b66f"
time="2024-06-27 15:55:59" level=info msg="Delete security group sg-0b344cc95f231b66f success "
time="2024-06-27 15:56:00" level=info msg="Release rules successfully for SG sg-09ed3a9ea9c1ab834"
time="2024-06-27 15:56:00" level=info msg="Delete security group sg-09ed3a9ea9c1ab834 success "

------------------------------
• [FAILED] [50.428 seconds]
Create machinepool [It] can create machinepool with additional security groups - [id:68173] [feature-machinepool, day2, High]
/Users/lixue/Workspace/rosa/tests/e2e/test_rosacli_machine_pool.go:467

  Timeline >>
  STEP: Get the cluster @ 06/27/24 15:55:10.388
  STEP: Init the client @ 06/27/24 15:55:10.389
  STEP: Skip testing if the cluster is not a Classic cluster @ 06/27/24 15:55:10.389
  time=2024-06-27T15:55:10+08:00 level=info msg=Running command: rosa describe cluster -c 2c58u55r4uh0cgf4omc1js5sakcei31i --output json
  time=2024-06-27T15:55:15+08:00 level=info msg=Get Combining Stdout and Stder is :
  {
    "additional_trust_bundle": "REDACTED",
    "addons": {
      "items": []
    },
    "api": {
      "listening": "external",
      "url": "https://api.xueli-g1wudq9et.lkuh.s1.devshift.org:6443"
    },
    "aws": {
      "additional_compute_security_group_ids": [
        "sg-0d6327eb0ce8c96ea",
        "sg-048f66d820441c2fa",
        "sg-0efb93b4b9cb18da2"
      ],
      "additional_control_plane_security_group_ids": [
        "sg-0d6327eb0ce8c96ea",
        "sg-048f66d820441c2fa",
        "sg-0efb93b4b9cb18da2"
      ],
      "additional_infra_security_group_ids": [
        "sg-0d6327eb0ce8c96ea",
        "sg-048f66d820441c2fa",
        "sg-0efb93b4b9cb18da2"
      ],
      "audit_log": {
        "role_arn": ""
      },
      "ec2_metadata_http_tokens": "required",
      "kms_key_arn": "arn:aws:kms:us-west-2:*************:key/7bfc89f3-8e2d-4767-b21d-0487b511dce7",
      "private_link": false,
      "private_link_configuration": {},
      "sts": {
        "auto_mode": false,
        "enabled": true,
        "instance_iam_roles": {
          "master_role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-ControlPlane-Role",
          "worker_role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-Worker-Role"
        },
        "managed_policies": false,
        "oidc_config": {
          "creation_timestamp": "2024-06-27T03:24:25Z",
          "href": "/api/clusters_mgmt/v1/oidc_configs/2c58qqasgrb2vcajis3o1qmtc9picgu9",
          "id": "2c58qqasgrb2vcajis3o1qmtc9picgu9",
          "issuer_url": "https://xueli-g1wudq9et-oidc-x2c9.s3.us-west-2.amazonaws.com",
          "last_update_timestamp": "2024-06-27T03:31:39Z",
          "last_used_timestamp": "2024-06-27T03:31:39Z",
          "managed": false,
          "organization_id": "1Ktle4i022EEYAi0K3y5rfWhHMM",
          "reusable": true,
          "secret_arn": "arn:aws:secretsmanager:us-west-2:*************:secret:rosa-private-key-xueli-g1wudq9et-oidc-x2c9-TXVAA0"
        },
        "oidc_endpoint_url": "https://xueli-g1wudq9et-oidc-x2c9.s3.us-west-2.amazonaws.com",
        "operator_iam_roles": [
          {
            "id": "",
            "name": "cloud-credentials",
            "namespace": "openshift-ingress-operator",
            "role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-openshift-ingress-operator-clou",
            "service_account": ""
          },
          {
            "id": "",
            "name": "ebs-cloud-credentials",
            "namespace": "openshift-cluster-csi-drivers",
            "role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-openshift-cluster-csi-drivers-e",
            "service_account": ""
          },
          {
            "id": "",
            "name": "cloud-credentials",
            "namespace": "openshift-cloud-network-config-controller",
            "role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-openshift-cloud-network-config-",
            "service_account": ""
          },
          {
            "id": "",
            "name": "aws-cloud-credentials",
            "namespace": "openshift-machine-api",
            "role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-openshift-machine-api-aws-cloud",
            "service_account": ""
          },
          {
            "id": "",
            "name": "cloud-credential-operator-iam-ro-creds",
            "namespace": "openshift-cloud-credential-operator",
            "role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-openshift-cloud-credential-oper",
            "service_account": ""
          },
          {
            "id": "",
            "name": "installer-cloud-credentials",
            "namespace": "openshift-image-registry",
            "role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-openshift-image-registry-instal",
            "service_account": ""
          }
        ],
        "operator_role_prefix": "xueli-g1wudq9etijigdhs8vg-uryyih",
        "role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-Installer-Role",
        "support_role_arn": "arn:aws:iam::*************:role/test/xueli-g1wudq9etijigdhs8vg-uryyih-Support-Role"
      },
      "subnet_ids": [
        "subnet-0b4f00f3560a65c23",
        "subnet-0a69c2da274bf20f1",
        "subnet-016492bd22837a647",
        "subnet-03cbf4f8fc135c687",
        "subnet-0091e54c2d71097d4",
        "subnet-050e8e0cf96f10d85"
      ],
      "tags": {
        "red-hat-clustertype": "rosa",
        "red-hat-managed": "true"
      }
    },
    "aws_infrastructure_access_role_grants": {
      "items": []
    },
    "billing_model": "standard",
    "byo_oidc": {
      "enabled": true
    },
    "ccs": {
      "disable_scp_checks": false,
      "enabled": true,
      "kind": "CCS"
    },
    "cloud_provider": {
      "href": "/api/clusters_mgmt/v1/cloud_providers/aws",
      "id": "aws",
      "kind": "CloudProviderLink"
    },
    "console": {
      "url": "https://console-openshift-console.apps.xueli-g1wudq9et.lkuh.s1.devshift.org"
    },
    "creation_timestamp": "2024-06-27T03:31:36Z",
    "delete_protection": {
      "enabled": false
    },
    "disable_user_workload_monitoring": true,
    "displayName": "xueli-g1wudq9etijigdhs8vg-uryyih3qkcjbojamuymhjvsdx3on",
    "dns": {
      "base_domain": "lkuh.s1.devshift.org"
    },
    "domain_prefix": "xueli-g1wudq9et",
    "etcd_encryption": true,
    "expiration_timestamp": "2024-06-28T15:31:32Z",
    "external_auth_config": {
      "enabled": false,
      "external_auths": {
        "items": []
      }
    },
    "external_configuration": {
      "labels": {
        "items": []
      },
      "manifests": {
        "items": []
      },
      "syncsets": {
        "items": []
      }
    },
    "external_id": "cafa2ab3-d0d5-4a77-85bb-ada8392c8eca",
    "fips": true,
    "flavour": {
      "href": "/api/clusters_mgmt/v1/flavours/osd-4",
      "id": "osd-4",
      "kind": "FlavourLink"
    },
    "groups": {
      "items": []
    },
    "href": "/api/clusters_mgmt/v1/clusters/2c58u55r4uh0cgf4omc1js5sakcei31i",
    "hypershift": {
      "enabled": false
    },
    "id": "2c58u55r4uh0cgf4omc1js5sakcei31i",
    "identity_providers": {
      "items": []
    },
    "inflight_checks": {
      "items": []
    },
    "infra_id": "xueli-g1wudq9et-4rc5k",
    "ingresses": {
      "items": []
    },
    "kind": "Cluster",
    "machine_pools": {
      "items": []
    },
    "managed": true,
    "managed_service": {
      "enabled": false
    },
    "multi_az": true,
    "name": "xueli-g1wudq9etijigdhs8vg-uryyih3qkcjbojamuymhjvsdx3on",
    "network": {
      "host_prefix": 25,
      "machine_cidr": "10.0.0.0/16",
      "pod_cidr": "192.168.0.0/18",
      "service_cidr": "172.31.0.0/24",
      "type": "OVNKubernetes"
    },
    "node_drain_grace_period": {
      "unit": "minutes",
      "value": 60
    },
    "nodes": {
      "autoscale_compute": {
        "kind": "MachinePoolAutoscaling",
        "max_replicas": 6,
        "min_replicas": 3
      },
      "availability_zones": [
        "us-west-2a",
        "us-west-2b",
        "us-west-2c"
      ],
      "compute_labels": {
        "test-label": "testvalue",
        "test-label/openshift.io": ""
      },
      "compute_machine_type": {
        "href": "/api/clusters_mgmt/v1/machine_types/r5.xlarge",
        "id": "r5.xlarge",
        "kind": "MachineTypeLink"
      },
      "compute_root_volume": {
        "aws": {
          "size": 150
        }
      },
      "infra": 3,
      "infra_machine_type": {
        "href": "/api/clusters_mgmt/v1/machine_types/r5.xlarge",
        "id": "r5.xlarge",
        "kind": "MachineTypeLink"
      },
      "master": 3
    },
    "openshift_version": "4.15.0-0.nightly-2024-06-27-024641",
    "product": {
      "href": "/api/clusters_mgmt/v1/products/rosa",
      "id": "rosa",
      "kind": "ProductLink"
    },
    "properties": {
      "rosa_cli_version": "1.2.41",
      "rosa_creator_arn": "arn:aws:iam::*************:user/xueli"
    },
    "proxy": {
      "http_proxy": "http://10.0.0.14:8080",
      "https_proxy": "https://10.0.0.14:8080",
      "no_proxy": "quay.io"
    },
    "region": {
      "href": "/api/clusters_mgmt/v1/cloud_providers/aws/regions/us-west-2",
      "id": "us-west-2",
      "kind": "CloudRegionLink"
    },
    "state": "ready",
    "status": {
      "configuration_mode": "full",
      "dns_ready": true,
      "kind": "ClusterStatus",
      "limited_support_reason_count": 0,
      "oidc_ready": true,
      "provision_error_code": "",
      "provision_error_message": "",
      "state": "ready"
    },
    "subscription": {
      "href": "/api/accounts_mgmt/v1/subscriptions/2iRh3GVgVIBE47DnRb2DBN8gEAN",
      "id": "2iRh3GVgVIBE47DnRb2DBN8gEAN",
      "kind": "SubscriptionLink"
    },
    "version": {
      "channel_group": "nightly",
      "end_of_life_timestamp": "2025-06-30T00:00:00Z",
      "href": "/api/clusters_mgmt/v1/versions/openshift-v4.15.0-0.nightly-2024-06-27-024641-nightly",
      "id": "openshift-v4.15.0-0.nightly-2024-06-27-024641-nightly",
      "kind": "Version",
      "raw_id": "4.15.0-0.nightly-2024-06-27-024641"
    }
  }

  STEP: Parse the cluster config @ 06/27/24 15:55:15.416
  STEP: Load the vpc client of the machinepool @ 06/27/24 15:55:15.417
  time=2024-06-27T15:55:15+08:00 level=info msg=Running command: rosa list machinepool -c 2c58u55r4uh0cgf4omc1js5sakcei31i
  time=2024-06-27T15:55:19+08:00 level=info msg=Get Combining Stdout and Stder is :
  ID      AUTOSCALING  REPLICAS  INSTANCE TYPE  LABELS                                            TAINTS    AVAILABILITY ZONES                    SUBNETS                                                                         SPOT INSTANCES  DISK SIZE  SG IDs
  worker  Yes          3-6       r5.xlarge      test-label=testvalue, test-label/openshift.io=              us-west-2a, us-west-2b, us-west-2c    subnet-016492bd22837a647, subnet-0a69c2da274bf20f1, subnet-0b4f00f3560a65c23    No              150 GiB    sg-0d6327eb0ce8c96ea, sg-048f66d820441c2fa, sg-0efb93b4b9cb18da2

  STEP: Prepare security groups @ 06/27/24 15:55:24.864
  STEP: Create machinepool with security groups set @ 06/27/24 15:55:31.573
  time=2024-06-27T15:55:31+08:00 level=info msg=Running command: rosa create machinepool --additional-security-group-ids sg-044700bb18e0be9a4,sg-0b344cc95f231b66f,sg-09ed3a9ea9c1ab834 --replicas 0 -y -c 2c58u55r4uh0cgf4omc1js5sakcei31i --name mp-68173
  time=2024-06-27T15:55:47+08:00 level=info msg=Get Combining Stdout and Stder is :
  INFO: Machine pool 'mp-68173' created successfully on cluster '2c58u55r4uh0cgf4omc1js5sakcei31i'
  INFO: To view the machine pool details, run 'rosa describe machinepool --cluster 2c58u55r4uh0cgf4omc1js5sakcei31i --machinepool mp-68173'
  INFO: To view all machine pools, run 'rosa list machinepools --cluster 2c58u55r4uh0cgf4omc1js5sakcei31i'

  STEP: Check the machinepool detail by describe @ 06/27/24 15:55:47.12
  time=2024-06-27T15:55:47+08:00 level=info msg=Running command: rosa describe machinepool mp-68173 -c 2c58u55r4uh0cgf4omc1js5sakcei31i
  time=2024-06-27T15:55:51+08:00 level=info msg=Get Combining Stdout and Stder is :

  ID:                                    mp-68173
  Cluster ID:                            2c58u55r4uh0cgf4omc1js5sakcei31i
  Autoscaling:                           No
  Replicas:                              0
  Instance type:                         m5.xlarge
  Labels:                                
  Taints:                                
  Availability zones:                    us-west-2a, us-west-2b, us-west-2c
  Subnets:                               subnet-016492bd22837a647, subnet-0a69c2da274bf20f1, subnet-0b4f00f3560a65c23
  Spot instances:                        No
  Disk size:                             300 GiB
  Additional Security Group IDs:         sg-044700bb18e0be9a4, sg-0b344cc95f231b66f, sg-09ed3a9ea9c1ab834
  Tags:                                  red-hat-clustertype=rosa, red-hat-managed=true

  time=2024-06-27T15:55:51+08:00 level=info msg=Running command: rosa delete machinepool -c 2c58u55r4uh0cgf4omc1js5sakcei31i mp-68173 -y
  time=2024-06-27T15:55:55+08:00 level=info msg=Get Combining Stdout and Stder is :
  INFO: Successfully deleted machine pool 'mp-68173' from cluster '2c58u55r4uh0cgf4omc1js5sakcei31i'

  [FAILED] in [It] - /Users/lixue/Workspace/rosa/tests/e2e/test_rosacli_machine_pool.go:503 @ 06/27/24 15:56:00.817
  STEP: Clean remaining resources @ 06/27/24 15:56:00.817
  << Timeline

  [FAILED] Expected
      <string>: 
  to equal
      <string>: sg-044700bb18e0be9a4, sg-0b344cc95f231b66f, sg-09ed3a9ea9c1ab834
  In [It] at: /Users/lixue/Workspace/rosa/tests/e2e/test_rosacli_machine_pool.go:503 @ 06/27/24 15:56:00.817
------------------------------
••SSS

Summarizing 1 Failure:
  [FAIL] Create machinepool [It] can create machinepool with additional security groups - [id:68173] [feature-machinepool, day2, High]
  /Users/lixue/Workspace/rosa/tests/e2e/test_rosacli_machine_pool.go:503

Ran 3 of 119 Specs in 149.884 seconds
FAIL! -- 2 Passed | 1 Failed | 0 Pending | 116 Skipped
--- FAIL: TestROSACLIProvider (149.89s)
FAIL

Ginkgo ran 1 suite in 2m36.63544464s

Test Suite Failed
```
The case id:68173 is failed due to a known issue.

File: tests/e2e/test_rosacli_machine_pool.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 
 var _ = Describe("Create machinepool",
 	labels.Feature.Machinepool,
+	labels.Runtime.Day2,
```
Comment (radtriste) [inline tests/e2e/test_rosacli_machine_pool.go:22]: Runtime should remain on the test case itself: https://github.com/openshift/rosa/blob/master/tests/README.md#labels

File: tests/e2e/test_rosacli_machine_pool.go [hunk 2]
```diff
@@ -461,6 +463,153 @@ var _ = Describe("Create machinepool",
...
+				mpName := "mp-68173"
+				_, err = rosaClient.MachinePool.CreateMachinePool(clusterID, mpName,
+					"--additional-security-group-ids", strings.Join(sgIDs, ","),
+					"--replicas", "0",
+					"-y",
+				)
+				Expect(err).ToNot(HaveOccurred())
+				defer rosaClient.MachinePool.DeleteMachinePool(clusterID, mpName)
+
+				By("Check the machinepool detail by describe")
```
Comment (radtriste) [inline tests/e2e/test_rosacli_machine_pool.go:499]:
```suggestion
				By("Check the machinepool details by describe")
```

**Review by radtriste — CHANGES_REQUESTED** [review]

File: tests/e2e/test_rosacli_machine_pool.go (hunk 1 above)
Comment (xueli181114) [reply tests/e2e/test_rosacli_machine_pool.go]: Updated

File: tests/e2e/test_rosacli_machine_pool.go (hunk 2 above)
Comment (xueli181114) [reply tests/e2e/test_rosacli_machine_pool.go]: Updated

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**